crates/sui-graphql-rpc/schema/draft_schema.graphql

```diff
@@ -191,6 +191,58 @@ scalar DateTime
 #   | { Struct:  [{ name: string, value: MoveData }] }
 scalar MoveData
 
+# The signature of a concrete Move Type (a type with all its type
+# parameters instantiated with concrete types, that contains no
+# references), corresponding to the following recursive type:
+#
+# type MoveTypeSignature =
+#     "address"
+#   | "bool"
+#   | "u8" | "u16" | ... | "u256"
+#   | { vector: MoveTypeSignature }
+#   | {
+#       package: string,
+#       module: string,
+#       type: string,
+#       typeParameters: [MoveTypeSignature]?,
+#     }
+scalar MoveTypeSignature
+
+# The shape of a concrete Move Type (a type with all its type
+# parameters instantiated with concrete types), corresponding to the
+# following recursive type:
+#
+# type MoveTypeLayout =
+#     "address"
+#   | "bool"
+#   | "u8" | "u16" | ... | "u256"
+#   | { vector: MoveTypeLayout }
+#   | { struct: [{ name: string, layout: MoveTypeLayout }] }
+scalar MoveTypeLayout
```

Comment:
is there a significant benefit to splitting `MoveTypeLayout` and `MoveTypeSignature` rather than combing them into one representation?

```
# type MoveTypeSignature =
#     "address"
#   | "bool"
#   | "u8" | "u16" | ... | "u256"
#   | { vector: MoveTypeSignature }
#   | {
#       package: string,
#       module: string,
#       type: string,
#       typeParameters: [MoveTypeSignature]?,
#       struct: [{ name: string, layout: MoveTypeLayout }]
#     }
```
I guess MoveTypeLayout is flat, so you could save nested lookups

Reply:
> I guess MoveTypeLayout is flat, so you could save nested lookups

Kind of -- it's the other way around though (i.e. it's `MoveTypeSignature` which is flat in this way), but I suspect that's what you meant?

Implementation-wise, `MoveTypeSignature` corresponds to `TypeTag` which represents the identity of a type, and then `MoveTypeLayout` is what you get when you recursively load the declarations of the types involved, so especially when they are scalars, it's beneficial to split them because the signature is free to look-up while the type layout is not.